### PR TITLE
KOGITO-366 DMN fix OCP asset resolution errors

### PR DESCRIPTION
**Pull request by tarilabs**
Requires https://github.com/kiegroup/kogito-runtimes/pull/181

The scope of this PR is to fix OCP resolution errors.

@ricardozanini you may want to use this one, regardless of the "warn" message, to check you don't occur any longer in the "multiple model same name error".

### CODE REVIEW

**Comment by ricardozanini** [comment]
@tarilabs I'll check it tomorrow, thanks!

**Comment by ricardozanini** [comment]
@tarilabs I'm facing this error when building on OCP with your branch:

```
INFO] Changes detected - recompiling the module!
[INFO] Compiling 6 source files to /tmp/src/target/classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /tmp/src/src/main/java/org/kie/kogito/examples/VacationDaysDMNEndpoint.java:[36,51] method createGenericDMNRuntime in class org.kie.kogito.dmn.DMNKogito cannot be applied to given types;
required: no arguments
found: java.io.InputStreamReader
reason: actual and formal argument lists differ in length
[ERROR] /tmp/src/src/main/java/org/kie/kogito/examples/PaymentDateDMNEndpoint.java:[35,51] method createGenericDMNRuntime in class org.kie.kogito.dmn.DMNKogito cannot be applied to given types;
required: no arguments
found: java.io.InputStreamReader
reason: actual and formal argument lists differ in length
[ERROR] /tmp/src/src/main/java/org/kie/kogito/examples/TaxRateDMNEndpoint.java:[36,51] method createGenericDMNRuntime in class org.kie.kogito.dmn.DMNKogito cannot be applied to given types;
required: no arguments
found: java.io.InputStreamReader
reason: actual and formal argument lists differ in length
[INFO] 3 errors
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  08:39 min
[INFO] Finished at: 2019-10-03T22:06:29Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0-jboss-1:compile (default-compile) on project payroll: Compilation failure: Compilation failure:
[ERROR] /tmp/src/src/main/java/org/kie/kogito/examples/VacationDaysDMNEndpoint.java:[36,51] method createGenericDMNRuntime in class org.kie.kogito.dmn.DMNKogito cannot be applied to given types;
[ERROR]   required: no arguments
[ERROR]   found: java.io.InputStreamReader
[ERROR]   reason: actual and formal argument lists differ in length
[ERROR] /tmp/src/src/main/java/org/kie/kogito/examples/PaymentDateDMNEndpoint.java:[35,51] method createGenericDMNRuntime in class org.kie.kogito.dmn.DMNKogito cannot be applied to given types;
[ERROR]   required: no arguments
[ERROR]   found: java.io.InputStreamReader
[ERROR]   reason: actual and formal argument lists differ in length
[ERROR] /tmp/src/src/main/java/org/kie/kogito/examples/TaxRateDMNEndpoint.java:[36,51] method createGenericDMNRuntime in class org.kie.kogito.dmn.DMNKogito cannot be applied to given types;
[ERROR]   required: no arguments
[ERROR]   found: java.io.InputStreamReader
[ERROR]   reason: actual and formal argument lists differ in length
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException 
error: build error: non-zero (13) exit code from 172.30.1.1:5000/openshift/kogito-quarkus-ubi8-s2i@sha256:ca691e5cda27758deed2090752e5b87f51a7f8c361cf5d7cbda044fce459d18d
```

I see some commits in the `bom` today, might be related? Thanks!

**Comment by tarilabs** [comment]
Looking at the error message looks like you are not having the required -runtimes changes applied correctly first

**Comment by ricardozanini** [comment]
> Looking at the error message looks like you are not having the required -runtimes changes applied correctly first

I just built your branch, which `-runtimes` changes I have to apply?

**Comment by tarilabs** [comment]
The https://github.com/kiegroup/kogito-runtimes/pull/181  is needed to avoid the compilation error you reported. It doesn't happen on Jenkins CI because indeed it takes into account changes from that branch

**Comment by tarilabs** [comment]
In other words, it might be "when on OCP" is not compiling using the changes in -runtimes required, at least the error message seems to indicate it is ignoring the changes in the linked -runtime PR (which at the time of writing is not merged)

**Comment by ricardozanini** [comment]
Let's wait for the -runtime changes to get merged to run this test again. Thanks!

File: onboarding-example/payroll/src/main/java/org/kie/kogito/examples/PaymentDateDMNEndpoint.java [hunk 1]
```diff
@@ -32,7 +32,7 @@
 @Path("/paymentDate")
 public class PaymentDateDMNEndpoint {
 
-    static final DMNRuntime dmnRuntime = DMNKogito.createGenericDMNRuntime();
+    static final DMNRuntime dmnRuntime = DMNKogito.createGenericDMNRuntime(new java.io.InputStreamReader(PaymentDateDMNEndpoint.class.getResourceAsStream("/org/kie/kogito/examples/payroll/paymentDate.dmn")));
```
Comment (mswiderski) [inline onboarding-example/payroll/src/main/java/org/kie/kogito/examples/PaymentDateDMNEndpoint.java:35]: Should we somehow handle closing of the reader?

**Review by mswiderski — APPROVED** [review]
Just one question about the reader closing

File: onboarding-example/payroll/src/main/java/org/kie/kogito/examples/PaymentDateDMNEndpoint.java (hunk 1 above)
Comment (tarilabs) [reply onboarding-example/payroll/src/main/java/org/kie/kogito/examples/PaymentDateDMNEndpoint.java:35]: IIRC it's done internally by the compiler, but I will double-check it's been a while

**Pull request merged**